src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -69,28 +72,45 @@ class PSyDataNode(Node):
     :type parent: :py:class:`psyclone.psyir.nodes.Node`
     :param options: a dictionary with options for transformations.
     :type options: dictionary of string:values or None
+    :param str options["class"]: a prefix to use for the PSyData module name \
+        (``prefix_psy_data_mod``) and the PSyDataType
+        (``prefix_PSyDataType``) - a "_" will be added automatically. \
+        It defaults to "", which means the module name used will just be \
+        ``psy_data_mode``, and the data type ``PSyDataType``.
     :param (str,str) options["region_name"]: an optional name to \
         use for this PSyDataNode, provided as a 2-tuple containing a \
         module name followed by a local name. The pair of strings should \
         uniquely identify a region unless aggregate information is required \
         (and is supported by the runtime library).
 
-
     '''
     # PSyData interface Fortran module
     fortran_module = "psy_data_mod"
     # The symbols we import from the PSyData Fortran module
     symbols = ["PSyDataType"]
-    # The use statement that will be inserted. Any use of a module of the
-    # same name that doesn't match this will result in a NotImplementedError
-    # at code-generation time.
-    use_stmt = "use psy_data_mod, only: " + ", ".join(symbols)
 
     # Root of the name to use for variables associated with PSyData regions
     psy_data_var = "psy_data"
 
     def __init__(self, ast=None, children=None, parent=None, options=None):
-        # TODO: #415 Support different classes of PSyData calls.
+
+        if not options:
+            options = {}
+
+        # This string stores a prefix to be used with all external PSyData
+        # symbols (i.e. data types and module name), used in the
+        # method 'make_symbol'.
+        self._class_string = options.get("class", "")
+        if self._class_string:
+            self._class_string = self._class_string+"_"
+
+        # The use statement that will be inserted. Any use of a module
+        # of the same name that doesn't match this will result in a
+        # NotImplementedError at code-generation time.
+        self.use_stmt = "use {0}, only: "\
```

Comment:
Besides this not being my code - I tried that, but imho the code gets then pretty complicated (UseGen to create the use statement, which also needs a parent, so you need a dummy node). Then the result of UseGen cannot be compared with the node, since UseGen is a different type. So you need UseGe(...).root to get an object of class Use. But then this STILL can not be compared with the node in the tree, since the node in the tree is of class Use_Stmt. So you need to convert the Use object and Use_Stmt object into strings. Which then leads to the question: why bother in the first place.  Am I doing something wrong??

The one option I do see is to use 
```
        parent = Node()
        u = UseGen(parent, name=self.make_symbol("psy_data_mod"),
                   only=True,
                   funcnames=[self.make_symbol(symbol) for symbol in
                      PSyDataNode.symbols])
        self.use_stmt = str(u.root)
```
to create the string? Is that what you meant?